pypesto/visualize/observable_mapping.py

```diff
@@ -37,10 +41,346 @@
+        )
+    )
+
+    ######################################
+    # Plot the linear observable mappings.
+    ######################################
+
+    # for each group, plot the linear observable mapping
+    for observable_index, observable_id in zip(
+        relative_observable_indices, relative_observable_ids
+    ):
+        # Get the ax for the current observable.
+        if rel_and_semiquant_obs_indices is not None:
+            ax_index = rel_and_semiquant_obs_indices.index(observable_index)
+        else:
+            ax_index = relative_observable_indices.index(observable_index)
+
+        ax = axes[ax_index]
+
+        # Get the inner parameters for the current observable.
+        inner_parameters = inner_problem.get_xs_for_group(observable_index + 1)
+
+        scaling_factor = None
+        offset = None
+
+        for inner_par in inner_parameters:
+            if inner_par.inner_parameter_type == InnerParameterType.SCALING:
+                scaling_factor = inner_par
+            elif inner_par.inner_parameter_type == InnerParameterType.OFFSET:
+                offset = inner_par
+
+        scaling_factor_value = (
+            inner_parameter_values[scaling_factor.inner_parameter_id]
+            if scaling_factor is not None
+            else 1
+        )
+        offset_value = (
+            inner_parameter_values[offset.inner_parameter_id]
+            if offset is not None
+            else 0
+        )
+
+        # Get the data mask for the current observable.
+        observable_data_mask = (
+            scaling_factor.ixs if scaling_factor is not None else offset.ixs
+        )
+
+        # Get the measurements for the current observable.
+        measurements = extract_expdata_using_mask(
+            expdata=inner_problem.data, mask=observable_data_mask
+        )
+
+        # Get the simulation for the current observable.
+        simulation = extract_expdata_using_mask(
+            expdata=sim, mask=observable_data_mask
+        )
+
+        # min_simulation = np.min(simulation)
+        # min_measurement = np.min(measurements)
+
+        # Plot the simulations on the x-axis and the measurements on the y-axis.
+        # ax.plot(simulation, np.full(len(simulation), min_measurement), "go", label="Model output")
+        # ax.plot(np.full(len(measurements), min_simulation), measurements, "bs", label="Measurements")
+
+        ax.plot(simulation, measurements, "bs", label="Measurements")
+
+        # Plot the linear mapping.
+        ax.plot(
+            np.sort(simulation),
+            scaling_factor_value * np.sort(simulation) + offset_value,
+            linestyle="-",
+            color="orange",
+            label="Linear mapping",
+        )
+
+        ax.legend()
+        ax.set_title(f"Observable {observable_id}")
+        ax.set_xlabel("Model output")
+        ax.set_ylabel("Measurements")
+
+        # Store the ax in the axes array.
+        axes[ax_index] = ax
```

Comment:
Above, `ax = axes[ax_index]` -- no need to overwrite the `axes`, right?

Reply:
Agreed